tcms/telemetry/api.py

```diff
@@ -115,3 +115,53 @@ def status_matrix(query=None):
         'data': data_set,
         'columns': columns
     }
+
+
+@rpc_method(name='Testing.execution_trends')
+def execution_trends(query=None):
+
+    if query is None:
+        query = {}
+
+    test_executions = TestExecution.objects.filter(**query).order_by('run_id')
+    pass_count = test_executions.filter(status__name=TestExecutionStatus.PASSED).count()
+
+    data_set = []
+    all_count = {
+        'pass': pass_count,
+        'other': test_executions.count() - pass_count
+    }
+    row = {'run_id': 0}
+    for test_execution in test_executions:
+        is_pass = test_execution.status.name == TestExecutionStatus.PASSED
+
+        if is_pass:
+            all_count['pass'] += 1
+        else:
+            all_count['other'] += 1
```

Comment:
Design your data structure with IDLE, PASS, FAIL, OTHER (generally speaking just use the status name).

Note: we should probably do some refactoring for these statuses as their meaning can vary from time to time and the most recent PRs need to somehow distinguish statuses without actually hard-coding their names.